matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/util/TemporaryFileCreator.kt

```diff
@@ -28,7 +28,10 @@ internal class TemporaryFileCreator @Inject constructor(
 ) {
     suspend fun create(): File {
         return withContext(Dispatchers.IO) {
-            File.createTempFile(UUID.randomUUID().toString(), null, context.cacheDir)
+            runCatching {
+                File.createTempFile(UUID.randomUUID().toString(), null, context.cacheDir)
+                        .apply { mkdirs() }
```

Comment:
I am not sure that `mkdirs()` here will have any effect while we store the files in the `cache directory`. (should always exists). 

Also its interesting that using cache directory (from the doc):

> Apps are strongly encouraged to keep their usage of cache space below the quota returned by StorageManager.getCacheQuotaBytes(java.util.UUID). If your app goes above this quota, your cached files will be some of the first to be deleted when additional disk space is needed. Conversely, if your app stays under this quota, your cached files will be some of the last to be deleted when additional disk space is needed.

So whatever we store there, there are cases that system will delete our files! Just an idea that this may cause some errors!

Reply:
It sounds weird to me too, but could it be possible that in some devices even the cache can run into the problem described here? Perhaps it's something we need to pay attention after we release this fix. The only other possible cause I can think of is the image is being acquired from the main thread, but to fix that is a long coroutine that covers lots of areas that really would be better without it.

Reply:
Yes, let's track it after the release with the changes